src/renderers/shaders/ShaderChunk/fog_fragment.glsl

```diff
@@ -1,22 +1,12 @@
 #ifdef USE_FOG
 
-	#ifdef USE_LOGDEPTHBUF_EXT
-
-		float depth = gl_FragDepthEXT / gl_FragCoord.w;
-
-	#else
-
-		float depth = gl_FragCoord.z / gl_FragCoord.w;
-
-	#endif
-
 	#ifdef FOG_EXP2
 
-		float fogFactor = whiteCompliment( exp2( - fogDensity * fogDensity * depth * depth * LOG2 ) );
+		float fogFactor = whiteCompliment( exp2( - fogDensity * fogDensity * fogDepth * fogDepth * LOG2 ) );
```

Comment:
Is exp2(something*LOG2) intended here? exp(something*LOG2) would give exp2(something), if exp2 is the base 2 exponential and LOG2 is the logarithm of 2. exp2(something*LOG2) then makes the exponentiation base, assuming something is the argument, equal to exp(log(2)*log(2)).

I also wonder why the density and depth are squared. What is the physical analogy?

Reply:
Would you like opening a new issue to discuss this?

Reply:
Yes, I think so (but not right now).

Consulting the source, `LOG2` is not the logarithm of 2, but the inverse of that. That means it converts a natural logarithm into a base 2 logarithm. So `exp2(something*LOG2)` is just a complicated way of writing `exp(something)`.

Reply:
I am opening an issue right now...

Reply:
#17264